packages/mirinae/src/inputs/dropdown/select-dropdown/PSelectDropdown.vue

```diff
@@ -1,3 +1,141 @@
+<script setup lang="ts">
+import {
+    computed,
+    reactive,
+    nextTick,
+    toRef,
+    ref,
+    useSlots,
+} from 'vue';
+
+import { onClickOutside } from '@vueuse/core';
+import { groupBy, reduce } from 'lodash';
+
+import PI from '@/foundation/icons/PI.vue';
+import { useProxyValue } from '@/hooks';
+import { useContextMenuFixedStyle } from '@/hooks/context-menu-fixed-style';
+import PIconButton from '@/inputs/buttons/icon-button/PIconButton.vue';
+import PContextMenu from '@/inputs/context-menu/PContextMenu.vue';
+import type { MenuItem } from '@/inputs/context-menu/type';
+import type { SelectDropdownSize } from '@/inputs/dropdown/select-dropdown/type';
+import {
+    SELECT_DROPDOWN_STYLE_TYPE,
+    CONTEXT_MENU_POSITION,
+    SELECT_DROPDOWN_SIZE,
+} from '@/inputs/dropdown/select-dropdown/type';
+
+
+interface Props {
+    useFixedMenuStyle?: boolean;
+    visibleMenu?: boolean;
+    invalid?: boolean;
+    loading?: boolean;
+    items?: MenuItem[];
+    selected?: string | number | MenuItem[];
+    disabled?: boolean;
+    indexMode?: boolean;
+    placeholder?: string;
+    styleType?: SELECT_DROPDOWN_STYLE_TYPE;
+    buttonIcon?: string;
+    menuPosition?: CONTEXT_MENU_POSITION;
+    readOnly?: boolean;
+    size?: SelectDropdownSize;
+    isFixedWidth?: boolean;
+}
+
+const props = withDefaults(defineProps<Props>(), {
+    useFixedMenuStyle: false,
+    visibleMenu: undefined,
+    isFixedWidth: false,
+    styleType: SELECT_DROPDOWN_STYLE_TYPE.DEFAULT,
+    disabled: false,
+    readOnly: false,
+    invalid: false,
+    loading: false,
+    menuPosition: CONTEXT_MENU_POSITION.LEFT,
+    placeholder: '',
+    items: () => [],
+    selected: undefined,
+    indexMode: false,
+    size: SELECT_DROPDOWN_SIZE.md,
+    buttonIcon: undefined,
+});
+
+const emit = defineEmits<{(e: 'update:selected', value?: string | number): void;
+    (e: 'select', value?: string | number): void;
+    (e: 'update:visible-menu', value?: boolean): void;
+    (e: 'focus-menu'): void;
+    (e: 'click-delete'): void;
+}>();
+
+const slots = useSlots();
+
+const state = reactive({
+    proxyVisibleMenu: useProxyValue<boolean | undefined>('visibleMenu', props, emit),
+    contextMenuRef: null as null|any,
```

Comment:
nitpick: Maybe this Ref would be not working..
use `toRef` or change to single `Ref` like `containerRef`.

Reply:
fixed it!